v2/internal/test/ca.go

```diff
@@ -113,13 +141,66 @@ func CreateCertificate(tb testing.TB, tmpl, parent *x509.Certificate, pub, priv
 	return cert
 }
 
+func CreateWebCredentials(t testing.TB) (*x509.CertPool, *tls.Certificate) {
+	ipaddresses := []net.IP{net.IPv4(127, 0, 0, 1), net.IPv6loopback}
+
+	rootCert, rootKey := CreateCACertificate(t, nil, nil,
+		WithIPAddresses(ipaddresses))
+
+	childCert, childKey := CreateX509Certificate(t, rootCert, rootKey,
+		WithIPAddresses(ipaddresses))
+
+	return x509util.NewCertPool([]*x509.Certificate{rootCert}),
+		&tls.Certificate{
+			Certificate: [][]byte{childCert.Raw},
+			PrivateKey:  childKey,
+		}
+}
+
 func NewSerial(tb testing.TB) *big.Int {
 	b := make([]byte, 8)
 	_, err := rand.Read(b)
 	require.NoError(tb, err)
 	return new(big.Int).SetBytes(b)
 }
 
+func WithSerial(serial *big.Int) CertificateOption {
+	return certificateOption(func(c *x509.Certificate) {
+		c.SerialNumber = serial
+	})
+}
+
+func WithValidityBounds(notBefore, notAfter time.Time) CertificateOption {
```

Comment:
nit: maybe something more terse? `WithLifetime`?

Reply:
I like it. I used ValidityBounds because of https://golang.org/src/crypto/x509/x509.go?s=21657:25704#L652